test/JoinTest.cpp

```diff
@@ -469,3 +510,256 @@ TEST(JoinTest, invalidJoinVariable) {
 
   ASSERT_ANY_THROW(Join(qec, valuesTree2, valuesTree, 0, 0));
 }
+
+// _____________________________________________________________________________
+TEST(JoinTest, joinTwoLazyOperationsWithAndWithoutUndefValues) {
+  auto performJoin =
+      [](std::vector<IdTable> leftTables, std::vector<IdTable> rightTables,
+         const IdTable& expected, bool expectPossiblyUndefinedResult,
+         ad_utility::source_location loc =
+             ad_utility::source_location::current()) {
+        auto l = generateLocationTrace(loc);
+        auto qec = ad_utility::testing::getQec();
+        RuntimeParameters().set<"lazy-index-scan-max-size-materialization">(0);
```

Comment:
Have you made sure to reset all runtimeParameters when you are done with them (asking for a friend...).
An easy way would be to implement a helper that is basically an RAII cleanup that stores the original value, sets the new value, and on destruction (the type should be `[[nodiscard]]` resets the value.